### PR TITLE
[WIP]:Debugging PR, for reference

### PR DESCRIPTION
**Pull request by henrykironde**
This PR is only for tests and reference.

### CODE REVIEW

**Comment by henrykironde** [comment]
I tried to test the package for release and I found out that we had skipped somethings. The changes we made are nice but they do not pass when you run the test locally with  ` R CMD build . `  and `R CMD check rdataretriever_2.0.3.tar.gz --as-cran`.

For some reason some tests fail, The two that did not surprise me due to the latest changes are related to version checks.
The other one which was unexpected is the `fetch` test. 
I did not debug any of the functions to see what was going on.

The other error is the manual fails to build. Apparently , it throws   `\examples lines wider than 100 characters:`

Running the check with the `--no-manual` flag will pass fine.

`R CMD check rdataretriever_2.0.3.tar.gz  --no-manual`

File: R/rdataretriever.R [hunk 1]
```diff
@@ -759,8 +758,19 @@ install_retriever <- function(method = "auto", conda = "auto") {
...
 retriever <- NULL
 
-.onLoad <- function(libname, pkgname) {
-  ## assignment in parent environment!
-  retriever <<- reticulate::import("retriever", delay_load = TRUE)
-  retriever_meets_min_version()
+.onLoad = function(libname, pkgname) {
+    if(reticulate::py_available()){
+        # install_retriever()
+        install_python_modules <- function(method = "auto", conda = "auto") {
```
Comment (henrykironde) [inline R/rdataretriever.R:764]: We removed this part of code but I think we may need to test if python is available and retriever is available.

**Comment by henrykironde** [comment]
Please feel free to take over this Pr and create a new one. 
For any windows tests we need to send the build source, `rdataretriever_2.0.3.tar.gz` to https://win-builder.r-project.org/
For testing purposes, you will need to add `cre` to your author description as `person("Henry", "Senyondo", role = c("aut", "cre"),` in the DESCRIPTION file before you upload the source.

**Comment by ethanwhite** [comment]
Thanks @henrykironde. Working on this now for a bit. I've got the issues with `semver` fixed (me being sloppy with imports).

**Pull request closed without merging**